File: solutions/reverse_string.py
#!/usr/bin/env python3
"""
Reverse String - solution
Reads all stdin and prints its reversal.
"""
import sys

def reverse_string(s: str) -> str:
    return s[::-1]

if __name__ == "__main__":
    data = sys.stdin.read()
    # Remove trailing newline added by some shells? keep exactly reversed input
    print(reverse_string(data), end='')

File: solutions/reverse_string.js
#!/usr/bin/env node
// Reverse String - Node.js solution
// Reads all stdin and writes reversed content to stdout

const fs = require('fs');

function reverseString(s) {
  return s.split('').reverse().join('');
}

if (require.main === module) {
  const input = fs.readFileSync(0, 'utf8');
  const out = reverseString(input);
  process.stdout.write(out);
}
